### PR TITLE
feat: Provide support for scaling apps using AWS SQS Queue to include in-flight messages as well

### PR DESCRIPTION
**Pull request by LiamStorkey**
Signed-off-by: Liam Storkey <liam.storkey@ip-10-0-0-12.ap-southeast-2.compute.internal>

- Added in a config option `scaleOnInFlight` which changes the formula used for the sqs queue scaling metrics
- default is set to true and the fomula is set to the as currently used.
- added in 2 new test cases

```
For the purposes of scaling, the default formula for "actual messages" is equal to `ApproximateNumberOfMessages` + `ApproximateNumberOfMessagesNotVislble`, since `NotVisible` in SQS terms means the message is still in-flight/processing. If you wish to only scale on `ApproximateNumberOfMessages` set `scaleOnInFlight` to `false`.
- `scaleOnInFlight` - Bool flag to set the metrics to scale on. When set to `false` "actual messages" is equal to `ApproximateNumberOfMessages`. When set to `true` "actual messages" is equal to `ApproximateNumberOfMessages` + `ApproximateNumberOfMessagesNotVislble`, since `NotVisible` in SQS terms means the message is still in-flight/processing. (default: true)
```

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *([PR](https://github.com/kedacore/keda-docs/pull/773))*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Relates to https://github.com/kedacore/keda-docs/pull/773
Fixes https://github.com/kedacore/keda/issues/3133

### CODE REVIEW

**Comment by LiamStorkey** [comment]
hey @[tomkerkhove](https://github.com/tomkerkhove) 
I have gotten all the test to pass (I'm trying to run e2e sooo fingers cossed that runs - see above comment with slash command)

It would be amazing if we could get some eyes on this 👍🏻 
Thanks

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,10 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
 
 ## Unreleased
 
+### Improvements
```
Comment (tomkerkhove) [inline CHANGELOG.md:37]: We already have an improvement section, would you mind adding it there please?

File: CHANGELOG.md [hunk 2]
```diff
@@ -34,6 +34,10 @@ To learn more about our roadmap, we recommend reading [this document](ROADMAP.md
...
+
+- **AWS SQS Queue:** Updated the sqs scaler to add the config flag `scaleOnInFlight`. ([#773](https://github.com/kedacore/keda-docs/pull/773))
```
Comment (tomkerkhove) [inline CHANGELOG.md:39]:
```suggestion
- **AWS SQS Queue:** Support for scaling to include in-flight messages. ([#3133](https://github.com/kedacore/keda/issues/3133))
```

**Review by tomkerkhove — APPROVED** [review]

**Review by zroubalik — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (LiamStorkey) [reply CHANGELOG.md]: I have updated the wording, links and moved it to the correct location 👍🏻

**Comment by tomkerkhove** [comment]
@JorTurFer Do you know what's up with this GH check?

**Comment by JorTurFer** [comment]
There is a new commit since e2e test was executed. Every commit enqueue the check again (like other CIs)

**Pull request merged**